### PR TITLE
review feat: Enable comments by default

### PR DESCRIPTION
**Pull request by surli**
This PR is a proposal to change a default behaviour in Spoon: until now, the default was to **not** take into account the comments.
I assume this behaviour was used because the comments were not properly parsed then. But I think it's not the case anymore and we got more users using Spoon for parsing comments: very recently we got a question on Stackoverflow linked to that: https://stackoverflow.com/questions/50780598/how-to-get-comments-using-spoon/ and I know we also got issues related to that.

So in the same way we decided to use `noclasspath` mode by default, I propose that we parse comments by default. WDYT?

### CODE REVIEW

**Comment by tdurieux** [comment]
The comment was not enable for performance reasons.

**Comment by surli** [comment]
> The comment was not enable for performance reasons.

Do we have an idea about the perf difference on a large code base?

**Comment by tdurieux** [comment]
We need to read again all the Java files.
It is no terribly slow but it has an impact.

**Comment by monperrus** [comment]
I'm rather OK for this new default behavior.

**Comment by monperrus** [comment]
THanks. We have to prepare a really good changelog with all the non-bacward compatible changes such as this one.

File: src/main/java/spoon/Launcher.java [hunk 1]
```diff
@@ -425,7 +432,19 @@ protected void processArguments() {
 		environment.setTabulationSize(jsapActualArgs.getInt("tabsize"));
 		environment.useTabulations(jsapActualArgs.getBoolean("tabs"));
 		environment.setCopyResources(!jsapActualArgs.getBoolean("no-copy-resources"));
-		environment.setCommentEnabled(jsapActualArgs.getBoolean("enable-comments"));
+
+		if (jsapActualArgs.getBoolean("enable-comments")) {
+			Launcher.LOGGER.warn("The option --enable-comments (-c) is deprecated as it is now the default behaviour in Spoon.");
+		} else {
+			Launcher.LOGGER.warn("Spoon now parse by default the comments. Consider using the option --disable-comments if you want the old behaviour.");
```
Comment (seintur) [inline src/main/java/spoon/Launcher.java]: Typo: parse -> parses

**Pull request merged**